### PR TITLE
initial Sidecar client impl

### PR DESCRIPTION
**Pull request by smiklosovic**
backup is not done yet

### CODE REVIEW

File: pkg/sidecar/client.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	return &BackupResponse{}, nil
+}
+
+/////////////////
+//// OPERATIONS
+/////////////////
+
+type DecommissionError struct {
+	status   int
+	msg      string
```
Comment (alourie) [inline pkg/sidecar/client.go]: I say let's keep the error itself here, not just the string

File: pkg/sidecar/client.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	GetStatus() (*StatusResponse, error)
+}
+
+func (client *Client) GetStatus() (*StatusResponse, error) {
+
+	statusResponse := &StatusResponse{}
+
+	response, err := client.restyClient.R().
+		SetHeader(ACCEPT, ApplicationJson).
+		SetResult(&statusResponse).
```
Comment (alourie) [inline pkg/sidecar/client.go]: same here, no need to use &

File: pkg/sidecar/client_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	} else {
+		return host
+	}
+}
+
+func TestSidecarClient_GetStatus(t *testing.T) {
+
+	client := NewSidecarClient(getHost(), &ClientOptions{Secure: false})
+
+	response, e := client.GetStatus()
```
Comment (alourie) [inline pkg/sidecar/client_test.go]: e -> err :-)

**Comment by benbromhead** [comment]
lgtm cc @zegelin

**Pull request merged**